crates/circuit/src/dag_circuit.rs

```diff
@@ -2239,34 +2240,33 @@ def _format(operand):
         Ok(if recurse {
```

Comment:
I wonder if we want an `&& CONTROL_FLOW_OP_NAMES.iter().any(|x| self.op_names.contains_key(&x.to_string()))` on this condition so we don't bother with the for loop if we know there are no control flow instructions. The same question applies to `size()` as well.

Reply:
That's a good idea, and allowed me to make the code a bit cleaner across both methods. I added a helper as well, which is private to `DAGCircuit` for now.

Done in af244c2f44c2268a7ca92c9419f7a7a80db0c06e.